### PR TITLE
Improve `diagnose-api-breaking-changes` error reporting

### PR DESCRIPTION
**Pull request by neonichu**
It looks like `swift-api-digester` is always returning 0, so failures to parse modules can turn into false negative breakage reports. Do a minimal sniff test of the produced JSON to see if it contains symbols and if it does not, we'll emit the full output of `swift-api-digester` which will hopefully contain any errors encountered.

### CODE REVIEW

File: Sources/Commands/Utilities/APIDigester.swift [hunk 1]
```diff
@@ -192,11 +193,24 @@ public struct SwiftAPIDigester {
...
         }
+
+        try self.fileSystem.readFileContents(outputPath).withData { data in
+            if let jsonObject = try JSONSerialization.jsonObject(with: data, options: []) as? [String:Any] {
+                guard let abiRoot = jsonObject["ABIRoot"] as? [String:Any] else {
+                    throw Error.failedToValidateBaseline(module: module)
+                }
+
+                guard let symbols = abiRoot["children"] as? NSArray, symbols.count > 0 else {
+                    throw Error.noSymbolsInBaseline(module: module, toolOutput: try result.utf8Output())
```
Comment (neonichu) [inline Sources/Commands/Utilities/APIDigester.swift:209]: cc @nkcsgexi, does seem like a valid way to check whether the produced file contains any symbols? Is there a better of figuring out whether `swift-api-digester` failed?

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Review by tomerd — APPROVED** [review]
lgtm, assuming this is the right way to know if the tool had an issue. seems unfortunate it does not return a non-zero code in such cases, so maybe something @nkcsgexi can help get fixed

**Comment by tomerd** [comment]
also: is there a test we can add, or would it be hard to reproduce the empty symbols file?

File: Sources/Commands/Utilities/APIDigester.swift (hunk 1 above)
Comment (nkcsgexi) [reply Sources/Commands/Utilities/APIDigester.swift:209]: Can we pass down `-compiler-style-diags` to swift-api-digester invocations? That flag should give us the correct return value.

**Comment by neonichu** [comment]
> also: is there a test we can add, or would it be hard to reproduce the empty symbols file?

Should be easy to do, this is basically happening with any error parsing the module, so any package using e.g. a non-standard search path should hit it. Will look into @nkcsgexi's suggestions and then create a test.

File: Sources/Commands/Utilities/APIDigester.swift (hunk 1 above)
Comment (neonichu) [reply Sources/Commands/Utilities/APIDigester.swift:209]: Still seeing a 0 return value here, even when passing `-compiler-style-diags`

**Comment by neonichu** [comment]
@swift-ci please smoke test
